Massachusetts Emergency Aid to the Elderly, Disabled and Children (EAEDC)

hua7450 opened the pull request:
Fixes #1712

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/age_threshold.yaml:
```diff
@@ -0,0 +1,12 @@
+metadata:
+  unit: year
+  period: year
+  label: Massachusetts EAEDA elderly age 
```
```suggestion
  label: Massachusetts EAEDA elderly age threshold
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/age_threshold.yaml:
```diff
@@ -0,0 +1,12 @@
+description: Massachusetts requires household have at least one elderly over is age for the Emergency Aid to the Elderly, Disabled and Children program.
```
```suggestion
description: Massachusetts limits the Emergency Aid to the Elderly, Disabled and Children program to households with at least one person above this age threshold.
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_net_income.py:
```diff
@@ -0,0 +1,42 @@
+        # dependent care deduction
+        dependent_care_deduction_eligible = add(spm_unit, period, ["ma_eaedc_dependent_care_deduction_eligible"])
+        # age = age of dependent 
+        dependent_care_expense = add(spm_unit, period,["child_support_expense"])
```
care_expenses

hua7450 replied:
Do we  dependent_care_expense in system? (care_expenses)

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_income_limit.py:
```diff
@@ -0,0 +1,34 @@
+    H = "H"  # Living Arrangement H
+    NONE = "None"
+
+
```
create a class:
```
class ma_eadc_living_arrange(Variable):
    value_type = Enum
    entity = SPMUnit
    possible_values = MassachusettsEAEDCLivingArrangement
    default_value = MassachusettsEAEDCLivingArrangement.NONE
    definition_period = YEAR
    label = 
    defined_for = StateCode.MA

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_income_limit.py:
```diff
@@ -0,0 +1,34 @@
+    NONE = "None"
+
+
+class ma_eaedc_income_limit(Variable):
```
make this a separate file

hua7450 commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_disabled_earned_income.py:
```diff
@@ -0,0 +1,13 @@
+
+class ma_eaedc_disabled_earned_income(Variable):
+    value_type = float
+    entity = SPMUnit
```
person level

hua7450 commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_earned_income.py:
```diff
@@ -0,0 +1,13 @@
+
+class ma_eaedc_earned_income(Variable):
+    value_type = float
+    entity = SPMUnit
```
person

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_disabled_earned_income.py:
```diff
@@ -0,0 +1,14 @@
+class ma_eaedc_disabled_earned_income(Variable):
+    value_type = float
+    entity = Person
+    label = "Massachusetts EAEDC disabled earned income"
```
```suggestion
    label = "Massachusetts EAEDC earned income of a disabled person"
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_total_earned_income.py:
```diff
@@ -0,0 +1,13 @@
+from policyengine_us.model_api import *
+
+
+class ma_eaedc_total_earned_income(Variable):
```
```suggestion
class ma_eaedc_no_disabled_earned_income(Variable):
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_earned_income.py:
```diff
@@ -0,0 +1,12 @@
+from policyengine_us.model_api import *
+
+
+class ma_eaedc_earned_income(Variable):
```
```suggestion
class ma_eaedc_total_earned_income(Variable):
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_total_earned_income.py:
```diff
@@ -0,0 +1,13 @@
+    definition_period = YEAR
+    defined_for = StateCode.MA
+
+    adds = ["ma_eaedc_earned_income"]
```
make a formula and lets do the same as in disabled income just with `~disabled & ~dependent`

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/ma_eaedc_dependent_care_deduction.py:
```diff
@@ -0,0 +1,28 @@
+        person = spm_unit.members
+        is_dependent = person("is_tax_unit_dependent", period)
+        age = person("age", period)
+        meets_age_limit = age < 18
```
lets parametrize this

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_dependent_care_deduction_eligible.py:
```diff
@@ -0,0 +1,18 @@
+    value_type = bool
+    entity = SPMUnit
+    label = "Eligible for Massachusetts EAEDC dependent care deduction"
+    unit = USD
```
```suggestion
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_disabled_income_eligible.py:
```diff
@@ -0,0 +1,20 @@
+    value_type = bool
+    entity = SPMUnit
+    label = "Disabled person's income eligible for Massachusetts EAEDC"
+    unit = USD
```
```suggestion
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_assets_eligible.py:
```diff
@@ -0,0 +1,21 @@
+    value_type = bool
+    entity = SPMUnit
+    label = "Assets eligible for Massachusetts EAEDC "
+    unit = USD
```
```suggestion
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_age_eligible.py:
```diff
@@ -0,0 +1,22 @@
+    value_type = bool
+    entity = SPMUnit
+    label = "Age eligible for Massachusetts EAEDC "
+    unit = USD
```
```suggestion
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_disabled_earned_income.py:
```diff
@@ -0,0 +1,14 @@
+    defined_for = StateCode.MA
+
+    def formula(person, period, parameters):
+        disabled = person("is_disabled", period)
```
add a is_dependent condition

MaxGhenis commented:
Can we complete this given its importance to MyFriendBen, or should we start with a new PR?

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/standard_assistance/increment.yaml:
```diff
@@ -0,0 +1,26 @@
+description: For additional household members, each will increase standard_assistance by this amount.
+metadata:
+  unit: currency-USD
```
```suggestion
  unit: currency-USD
  period: year
```

PavelMakarchuk replied:
period in yaml files

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/standard_assistance/increment.yaml:
```diff
@@ -0,0 +1,26 @@
+description: For additional household members, each will increase standard_assistance by this amount.
```
```suggestion
description: Massachusetts increases the standard assistance for each additional household member by this amount, based on the living arrangement, under the Emergency Aid to the Elderly Disabled and Children program.
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/standard_assistance/base.yaml:
```diff
@@ -0,0 +1,26 @@
+description: Massachusetts assign these values as the standard_assistance for households size of 1. 
+metadata:
+  unit: /1
```
```suggestion
  unit: currency-USD
```

PavelMakarchuk replied:
if monthly, add period: month

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/standard_assistance/base.yaml:
```diff
@@ -0,0 +1,26 @@
+description: Massachusetts assign these values as the standard_assistance for households size of 1. 
```
```suggestion
description: Massachusetts provides the following standard assistance amount, based on living arrangements under the Emergency Aid to the Elderly Disabled and Children program.
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_standard_assistance.py:
```diff
@@ -0,0 +1,21 @@
+        n = spm_unit("spm_unit_size", period)
+        living_arrangement = spm_unit("ma_eaedc_living_arrangement", period)
+        p = parameters(period).gov.states.ma.dta.tcap.eaedc.standard_assistance
+        p1 = p.base[living_arrangement]
```
if we change the parameter to monthly, then use `MONTHS_IN_YEAR`

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/assets/limit.yaml:
```diff
@@ -0,0 +1,13 @@
+metadata:
+  threshold_unit: person 
+  amount_unit: currency-USD
+  type: single_amount
```
```suggestion
```

PavelMakarchuk replied:
Still need to adjust

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/assets/limit.yaml:
```diff
@@ -0,0 +1,13 @@
+  2022-01-01: 2_000
+
+metadata:
+  threshold_unit: person 
+  amount_unit: currency-USD
```
```suggestion
  unit: currency-USD
  period: year
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/assets/limit.yaml:
```diff
@@ -0,0 +1,13 @@
+  threshold_unit: person 
+  amount_unit: currency-USD
+  type: single_amount
+  threshold_period: year
```
```suggestion
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/dependent_care/maximum_expense.yaml:
```diff
@@ -0,0 +1,20 @@
+      2022-01-01: 2
+    amount:
+      2022-01-01: 200
+  # the legal code said this amount per child 
```
```suggestion
  # Legal code provides amount per child
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/income_disregards/amount.yaml:
```diff
@@ -0,0 +1,11 @@
+
+metadata:
+  unit: currency-USD
+  period: year
```
```suggestion
  period: month
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/income/disabled_cap.yaml:
```diff
@@ -0,0 +1,11 @@
+  
+metadata:
+  unit: year
+  period: year
```
```suggestion
  period: month
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/work_related_expense.yaml:
```diff
@@ -0,0 +1,11 @@
+  
+metadata:
+  unit: currency-USD
+  period: year
```
```suggestion
  period: month
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_eligible.py:
```diff
@@ -0,0 +1,28 @@
+    value_type = bool
+    entity = SPMUnit
+    label = "Eligible for Massachusetts EAEDC"
+    unit = USD
```
```suggestion
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_eligible.py:
```diff
@@ -0,0 +1,28 @@
+class ma_eaedc_eligible(Variable):
+    value_type = bool
+    entity = SPMUnit
+    label = "Eligible for Massachusetts EAEDC"
```
```suggestion
    label = "Eligible for the Massachusetts EAEDC"
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_financially_eligible.py:
```diff
@@ -0,0 +1,16 @@
+    value_type = bool
+    entity = SPMUnit
+    label = "Financially eligible for Massachusetts EAEDC "
+    unit = USD
```
```suggestion
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/ma_eaedc.py:
could you add this to the `household_state_benefits.yaml`

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/assets/list.yaml:
lets rename this to `sources.yaml`

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/dependent_care/age_threshold.yaml:
```diff
@@ -0,0 +1,12 @@
+  2021-01-01: 18
+  
+metadata:
+  unit: age
```
```suggestion
  unit: year
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/dependent_care/maximum_expense.yaml:
```diff
@@ -0,0 +1,20 @@
+description: Massachusetts provides this deduction amount in dependent care expense for Emergency Aid to the Elderly, Disabled and Children program.
```
```suggestion
description: Massachusetts provides this deduction amount of dependent care expenses for Emergency Aid to the Elderly, Disabled and Children program, based on the child age.
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/income_disregards/months.yaml:
```diff
@@ -0,0 +1,11 @@
+
+metadata:
+  unit: month
+  period: month
```
```suggestion
  period: year
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/income/disabled_cap.yaml:
```diff
@@ -0,0 +1,11 @@
+description: Massachusetts limits disabled peoople's income to this amount for the Emergency Aid to the Elderly, Disabled and Children program. 
```
```suggestion
description: Massachusetts limits the Emergency Aid to the Elderly, Disabled and Children program to disabled filers with income below this threshold. 
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/income/disabled_cap.yaml:
lets call this `disabled_limit.yaml`

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/standard_assistance/base.yaml:
```diff
@@ -0,0 +1,27 @@
+metadata:
+  unit: currency-USD  
+  period: month
+  label: Massachusetts EAEDC standard_assistance base
```
```suggestion
  label: Massachusetts EAEDC standard assistance base 
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/standard_assistance/increment.yaml:
```diff
@@ -0,0 +1,28 @@
+B:
+  2022-01-01: 121
+C:
+  2022-01-01: 0 #or 72.8
```
What does the OR stand for?

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/standard_assistance/increment.yaml:
```diff
@@ -0,0 +1,28 @@
+metadata:
+  unit: currency-USD
+  period: month
+  label: Massachusetts EAEDC standard_assistance increment 
```
```suggestion
  label: Massachusetts EAEDC standard assistance additional 
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/standard_assistance/base.yaml:
lets make an `amount` folder with the files:
`base.yaml`
`additional.yaml`

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_assets_eligible.py:
```diff
@@ -0,0 +1,18 @@
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.ma.dta.tcap.eaedc.assets
+        countable_assets = add(spm_unit, period, ["ma_eaedc_assets"])
```
```suggestion
        countable_assets = spm_unit("ma_eaedc_assets", period)
```

PavelMakarchuk replied:
we can just do this if `ma_eaedc_assets` is spm unit level

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_assets_eligible.py:
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class ma_eaedc_assets_eligible(Variable):
```
```suggestion
class ma_eaedc_asset_limit_eligible(Variable):
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_disabled_income_eligible.py:
Do we check this for the entire unit or just for the Disabled filers?

PavelMakarchuk replied:
If we want to do an spm unit level check then we need to add a `spm_unit.any` to the return 

Or change this to be a person level variable

PavelMakarchuk replied:
This is breaking the git test

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/me_eaedc_no_disabled_earned_income.py:
```diff
@@ -0,0 +1,19 @@
+from policyengine_us.model_api import *
+
+
+class ma_eaedc_no_disabled_earned_income(Variable):
```
```suggestion
class ma_eaedc_non_disabled_earned_income(Variable):
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_disabled_income_eligible.py:
```diff
@@ -0,0 +1,24 @@
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.ma.dta.tcap.eaedc.income
+        person = spm_unit.members
+        disabled_income = person("ma_eaedc_disabled_earned_income", period)
+        has_disabled = spm_unit.any(person("is_disabled", period))
+
+        return where(
+            has_disabled,
+            spm_unit.all(disabled_income < p.disabled_limit),
+            True,
+        )
```
```suggestion
        disabled_income = add(spm_unit, period, ["ma_eaedc_disabled_earned_income"])
        return disabled_income < p.disabled_limit
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_disabled_income_eligible.py:
```diff
@@ -0,0 +1,24 @@
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.ma.dta.tcap.eaedc.income
+        person = spm_unit.members
```
```suggestion
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/age_threshold.yaml:
```diff
@@ -0,0 +1,12 @@
+description: Massachusetts limits the Emergency Aid to the Elderly, Disabled and Children program to households with at least one person above this age threshold. 
```
```suggestion
description: Massachusetts limits the Emergency Aid to the Elderly, Disabled and Children program to households with at least one person at or above this age threshold. 
```

right?

MaxGhenis replied:
`this age or older` let's please use this phrasing for age-related parameters

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/assets/limit.yaml:
```diff
@@ -0,0 +1,11 @@
+description: Massachusetts limits the assets to this amount under the Emergency Aid to the Elderly, Disabled and Children program, for living arrangements E.
```
```suggestion
description: Massachusetts limits the assets to this amount under the Emergency Aid to the Elderly, Disabled and Children program, for individuals in living arrangement E.
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/dependent_care/age_threshold.yaml:
```diff
@@ -0,0 +1,12 @@
+description: Massachusetts limits the Emergency Aid to the Elderly, Disabled and Children program to households with at least one person above this age threshold. 
```
This defines a dependent right? 

Can we re-frame this dependent age threshold and add a parameter for the 19 year olds for are stundents

PavelMakarchuk replied:
```suggestion
description: Massachusetts defines a dependent as a person under this age threshold under the Emergency Aid to the Elderly, Disabled and Children program. 
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/dependent_care/maximum_expense.yaml:
```diff
@@ -0,0 +1,20 @@
+description: Massachusetts provides this deduction amount of dependent care expenses for Emergency Aid to the Elderly, Disabled and Children program, based on the child age.
```
```suggestion
description: Massachusetts provides this dependent care expense deduction amount under the Emergency Aid to the Elderly, Disabled and Children program, based on the child age.
```

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/dependent_care/maximum_expense.yaml:
we also should also add a partial deduction amount based on weekly hours worked:
![image](https://github.com/user-attachments/assets/333aa7bf-a11d-4c43-97b4-09f65cae64f4)

Two parameters one for younger and one for older

PavelMakarchuk commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/dependent_care/age_threshold.yaml:
```diff
@@ -0,0 +1,12 @@
+metadata:
+  unit: year
+  period: year
+  label: Massachusetts EAEDA elderly age threshold
```
```suggestion
  label: Massachusetts EAEDA elderly dependent age threshold
```

MaxGhenis replied:
Not EAEDC?

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_disabled_income_eligible.py:
```diff
@@ -0,0 +1,19 @@
+class ma_eaedc_disabled_income_eligible(Variable):
+    value_type = bool
+    entity = SPMUnit
+    label = "Disabled person's income eligible for the Massachusetts EAEDC"
```
```suggestion
    label = "Disability income eligibility for the Massachusetts EAEDC"
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_financially_eligible.py:
```diff
@@ -0,0 +1,15 @@
+from policyengine_us.model_api import *
+
+
+class ma_eaedc_financially_eligible(Variable):
```
```suggestion
class ma_eaedc_income_eligible(Variable):
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/eligibility/ma_eaedc_financially_eligible.py:
```diff
@@ -0,0 +1,15 @@
+class ma_eaedc_financially_eligible(Variable):
+    value_type = bool
+    entity = SPMUnit
+    label = "Financially eligible for the Massachusetts EAEDC "
```
```suggestion
    label = "Eligible for the Massachusetts EAEDC based on income"
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/ma_eaedc_disabled_earned_income.py:
```diff
@@ -0,0 +1,19 @@
+class ma_eaedc_disabled_earned_income(Variable):
+    value_type = float
+    entity = Person
+    label = "Massachusetts EAEDC earned income of a disabled person"
```
```suggestion
    label = "Massachusetts EAEDC earned income of each disabled person"
```

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/income/me_eaedc_non_disabled_earned_income.py:
```diff
@@ -0,0 +1,19 @@
+        dependent = person("is_tax_unit_dependent", period)
+        return (
+            ~disabled
+            * ~dependent
```
is this specifically for non-dependents? Or any non-disabled person

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/ma_eaedc.py:
```diff
@@ -0,0 +1,15 @@
+    defined_for = "ma_eaedc_eligible"
+    reference = "https://www.law.cornell.edu/regulations/massachusetts/106-CMR-704-500#(B)"
+
+    adds = ["ma_eaedc_standard_assistance"]
```
```suggestion
    # Program value can not be less than 0
    # due to the eligibility requirements 
    adds = ["ma_eaedc_standard_assistance"]
```

Right?

PavelMakarchuk commented on policyengine_us/variables/gov/states/ma/dta/tcap/eaedc/ma_eaedc_assets.py:
```diff
@@ -0,0 +1,12 @@
+from policyengine_us.model_api import *
+
+
+class ma_eaedc_assets(Variable):
```
```suggestion
class ma_eaedc_countable_assets(Variable):
```

PavelMakarchuk reviewed (changes requested):
Last formatting comments

MaxGhenis commented:
Per our chat just now, please align the disabled earned income test logic closer to the regs. We also need to populate the household type enum though could be in a separate PR.

hua7450 commented:
Here is what I did in the last commit:
 1. Restructured eligibility logic - created ma_eaedc_disabled_dependent_criteria_met variable
 2. Renamed ma_eaedc_disabled_eligible to ma_eaedc_head_is_disabled for clarity
 3. Refactored income calculation by extracting logic into separate ma_eaedc_earned_net_income variable
 4. Fixed vectorization in test files to return arrays instead of scalars
 5. Added new test cases to cover the updated functionality

MaxGhenis commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/dependent_care/maximum_deductions/older.yaml:
```diff
@@ -0,0 +1,28 @@
+description: Massachusetts provides this dependent care expense deduction amount under the Emergency Aid to the Elderly, Disabled and Children program, based on the child age.
```
```suggestion
description: Massachusetts provides this dependent care expense deduction amount under the Emergency Aid to the Elderly, Disabled and Children program, based on the child's age.
```

MaxGhenis commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/deductions/work_related_expense.yaml:
```diff
@@ -0,0 +1,11 @@
+description: Massachusetts provides this deduction amount in work-related expense for Emergency Aid to the Elderly, Disabled and Children program.
```
```suggestion
description: Massachusetts deducts up to this amount in work-related expenses for the Emergency Aid to the Elderly, Disabled and Children program.
```

MaxGhenis reviewed (changes requested):
please apply this feedback throughout

hua7450 commented on policyengine_us/parameters/gov/states/ma/dta/tcap/eaedc/README.md:
Yes https://www.law.cornell.edu/regulations/massachusetts/106-CMR-701-050

MaxGhenis commented:
Looks like we're adding the dependent care deduction already but just adding some info from Jenn Superka:

> I’m following up on the question about whether EAEDC applicants can deduct child care expenses in determining eligibility. Pavel, you had pointed out that the regs were recently updated, which I found (EAEDC was under 106 CMR 320, now it’s combined with TAFDC under 704). According to [106 CMR 704.500](https://www.mass.gov/doc/106-cmr-704-transitional-cash-assistance-program-financial-eligibility/download): Calculation of Grant Amount (and the [DTA website](https://www.mass.gov/info-details/how-to-calculate-eaedc-benefits)), when calculating eligibility for EAEDC, child care expenses are an allowable deduction.
> As you noted, there is a provision in 106 CMR 704.275(B) that states that an EAEDC applicant who meets the provisions of 106 CMR 704.285(C) is not eligible for the dependent care deduction. But there is no subsection C in 704.285.
> I reached out to one of our contacts in MA to ask about this missing provision, but it looks to me that the child care deduction is allowed.

MaxGhenis reviewed (commented):
@PavelMakarchuk and I just discussed making this monthly
